detectindent.lua

```diff
@@ -5,19 +5,50 @@ local util = import("micro/util")
 
 function onBufferOpen(buf)
     local spaces, tabs = 0, 0
+    local space_count, prev_space_count = -1, -1
+    local tabsizes = {}
     local i = 0
     while spaces + tabs < 500 and i < 1000 and i < buf:LinesNum() do
-        local r = util.RuneAt(buf:Line(i), 0)
+        space_count = -1
+        local line = buf:Line(i)
+        local r = util.RuneAt(line, 0)
         if r == " " then
             spaces = spaces + 1
+            space_count = string.len(util.GetLeadingWhitespace(line))
+            -- count empty lines as having the same indentation as the last one
+            if string.len(line) == space_count then
```

Comment:
Since this check is done inside `if r == " "`, it doesn't work for empty lines, only for blank lines consisting of 1 or more spaces.

Reply:
Good catch, I've added a separate test for empty lines.